src/core/server/plugins/plugins_system.ts

```diff
@@ -167,21 +169,40 @@ export class PluginsSystem {
 
   private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
     // make _cat/plugins?format=json call to the OpenSearch instance
-    const opensearchPlugins = await this.getOpenSearchPlugins(deps);
+    const opensearchInstalledPlugins = await this.getOpenSearchPlugins(deps);
     for (const pluginName of this.satupPlugins) {
       this.log.debug(`For plugin "${pluginName}"...`);
       const plugin = this.plugins.get(pluginName)!;
-      const pluginBackendDeps = new Set([...plugin.requiredOpenSearchPlugins]);
-      pluginBackendDeps.forEach((opensearchPlugin) => {
-        if (!opensearchPlugins.find((obj) => obj.component === opensearchPlugin)) {
+      const pluginOpenSearchDeps = Object.entries(plugin.requiredEnginePlugins);
+      for (const [enginePluginName, versionRange] of pluginOpenSearchDeps) {
+        // add check to see if the installing Dashboards plugin version is compatible with installed OpenSearch plugin
+        if (
+          !this.isVersionCompatibleOSPluginInstalled(
+            opensearchInstalledPlugins,
+            enginePluginName,
+            versionRange
+          )
+        ) {
           this.log.warn(
-            `OpenSearch plugin "${opensearchPlugin}" is not installed on the cluster for the OpenSearch Dashboards plugin to function as expected.`
+            `OpenSearch plugin "${enginePluginName}" is not installed on the cluster for the OpenSearch Dashboards plugin to function as expected.`
```

Comment:
We say "engine" elsewhere and "cluster" here; are they referring to the same thing?

Reply:
Yes they refer to same thing but its good idea to call it engine here to avoid confusion.